### PR TITLE
Feature/unsubscribe

### PR DESCRIPTION
**Pull request by ric9176**
This PR allows users to unsubscribe from our comms automatically rather than manually.

### CODE REVIEW

File: src/pages/unsubscribe.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  handleFormSubmit = e => {
+    e.preventDefault()
+    this.setState({ formSubmited: true })
+    this.triggerUnsubscribe()
+  }
+
+  triggerUnsubscribe = () => {
+    fetch(
```
Comment (alexlbr) [inline src/pages/unsubscribe.js]: @ric9176 why don't you move the fetch into a different file, then add the fetch as a prop (https://github.com/reactjsacademy/fb-messenger/blob/testing-react-leanjs/src/components/Messenger/Conversation/Content/Messages.js#L125), and write a test for it? This looks like an important functionality

File: src/pages/unsubscribe.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }),
+      }
+    )
+  }
+
+  handleEmailChange = e => {
+    this.setState({ email: e.target.value }, this.validateEmail)
+  }
+
+  validateEmail = () => {
```
Comment (alexlbr) [inline src/pages/unsubscribe.js]: @ric9176 why don't we use final-form to keep the state of the input and validate it? it makes the code simpler and we are already using final-form in this project

File: src/pages/unsubscribe.js (hunk 1 above)
Comment (ric9176) [reply src/pages/unsubscribe.js]: Yes I agree, we should def do that :+1: was just being fast as this is the mvp implementation

File: src/pages/unsubscribe.js (hunk 2 above)
Comment (ric9176) [reply src/pages/unsubscribe.js]: good point. I'll make that refactor, me and @horacioh did this quick version last night

Comment (horacioh) [reply src/pages/unsubscribe.js]: are you going to do this changes @ric9176 ?

Comment (ric9176) [reply src/pages/unsubscribe.js]: @horacioh I've done and pushed here the first change already if you want to do this one (Final form)

Comment (horacioh) [reply src/pages/unsubscribe.js]: cool!

File: src/components/buttons/LinkButton.js [hunk 3]
```diff
@@ -24,9 +24,6 @@ const fontColor = color => `
 `
 
 const StyledLinkButton = styled(Link)`
-svg {
```
Comment (ric9176) [inline src/components/buttons/LinkButton.js]: Why remove this? I think it is there bc we pass an svg as a child to LinkButton in ScholarshipsCard (tho not in use atm)

Comment (horacioh) [reply src/components/buttons/LinkButton.js]: thats a wrong css declararion, inside there it was only a value but no css valid declararion.

if you tell me whats the correct css declaration we can add it again :)

File: src/components/buttons/LinkButton.js [hunk 4]
```diff
@@ -82,10 +79,12 @@ const LinkButton = ({
 
 LinkButton.propTypes = {
   trackUserBehaviour: PropTypes.func,
+  secondary: PropTypes.bool
 }
 
 LinkButton.defaultProps = {
   trackUserBehaviour,
+  secondary: false
```
Comment (ric9176) [inline src/components/buttons/LinkButton.js]: ahhh this is why the snapShot is different on the CI, bc now markeingCard which uses LinkButton will get this default prop ```secondary={false}```  Anyway I'll update the snapshot

Comment (horacioh) [reply src/components/buttons/LinkButton.js]: nice!! I think we can remove it, its better without i guess..

**Comment by ric9176** [comment]
@alexlbr I think this one is good to merge now if those changes look good to you

File: src/api/triggerUnsubscribe.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const triggerUnsubscribe = email => {
```
[the remark this answers is not in the document]
Comment (ric9176) [reply src/api/triggerUnsubscribe.js]: :+1:

File: src/api/triggerUnsubscribe.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  fetch(
+    `https://us-central1-reactjsacademy-react.cloudfunctions.net/unsubscribe`,
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email: email,
```
[the remark this answers is not in the document]
Comment (ric9176) [reply src/api/triggerUnsubscribe.js]: :+1:

File: src/pages/unsubscribe.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+` //TODO: animate this later
+
+class Unsubscribe extends React.Component {
+  state = {
+    formSubmited: false
+  }
+
+  handleFormSubmit = ({ email }) => {
+    this.setState({ formSubmited: true })
+    this.props.api.triggerUnsubscribe(email)
```
[the remark this answers is not in the document]
Comment (ric9176) [reply src/pages/unsubscribe.js]: :100 I spoke with @horacioh about the test yesterday as I was working on other site updates, we are going to do a quick pairing on it this week and record video as cloud functions and the reason for the implementation is interesting (we think)

File: src/api/index.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email,
+      }),
+    }
+  )
+
+  export const triggerSubscribe = ({email, pathname}) => {
+    window &&
```
[the remark this answers is not in the document]
Comment (horacioh) [reply src/api/index.js]: question: is there a possible case where `window` will not exist?? remember that this api method is being called on a form submit. do we need this check??

**Pull request merged**